compose/service.py

```diff
@@ -402,9 +412,13 @@ def config_hash(self):
         return json_hash(self.config_dict())
 
     def config_dict(self):
+        image = self.image()
+        if image is None:
```

Comment:
Is there some context why you're raising it here rather than at source within: https://github.com/aanand/fig/blob/fix-smart-recreate-nonexistent-image/compose/service.py#L241

My thinking is other functions relying on image() returning a valid image won't benefit from this raise of `NoSuchImage`